### PR TITLE
Simpler dependency handling for micropip

### LINKED ISSUE #1619: Strange behavior with importing seaborn !
Hello! 
Someone ran into issues importing seaborn through a starboard notebook and, at first it seemed like a simple issue of wheel availability and versions. 
But in trying to find a specific version that maybe works, something just doesn't seem right. 

Ultimately I think that 
micropip.install('seaborn==0.9.0') worked in pyodide 0.16 but it doesn't seem to work in 0.17, and I can't quite figure out why! 

The post + my original response is here, for further context: 
https://community.starboard.gg/t/issue-with-micropip-installing-packages/100

### LINKED ISSUE #1745: Fix micropip dependency resolution for old package versions
Micropip does not correctly handle dependency resolution for old package versions #1619. It is difficult to fix because collecting the dependencies for old versions of a package from pypi is not that easy. pip uses a complicated scheme involving range requests, but pypi does not set CORS headers on responses to range requests pypa/conveyor#11 so we cannot use the same approach.

I opened another issue about this pypa/warehouse#9835 suggesting that they should add some simpler method of getting requirements of old package versions. This would allow improvements to pip too.

I think in the meantime since #1596 has been merged we should consider moving to a slower but correct approach and caching dependencies in IDBFS if possible. Perhaps other methods of caching package dependencies could also be explored.

### PR DESCRIPTION
**Pull request by hoodmane**
This is a simpler alternative to #1753 that skips the whole `LazyWheel` thing and just downloads the whole wheel directly. Hopefully we can merge this for v0.18 and then assess the merits of `LazyWheel` / #1753 separately after the v0.18 release.

Closes #1619
Closes #1745

### CODE REVIEW

**Comment by hoodmane** [comment]
This is around 2x faster than `LazyWheel` (installing sphinx takes about 10 seconds with `LazyWheel` and 5 seconds this way and on master). I wonder if it is even worth using `LazyWheel` for doing dependency resolution, probably it's better to make fewer requests and not worry about downloading unneeded data.

**Comment by hoodmane** [comment]
Yet again `install-custom-url` is failing.

File: packages/micropip/src/micropip/micropip.py [hunk 1]
```diff
@@ -222,13 +239,20 @@ async def add_requirement(self, requirement: str, ctx, transaction):
...
+    async def add_wheel(self, name, wheel, version, extras, ctx, transaction):
+        transaction["locked"][name] = version
+        response = await fetch(wheel["url"])
+        wheel_bytes = (await response.arrayBuffer()).to_py()
+        wheel["wheel_bytes"] = wheel_bytes
 
-        recurs_reqs = metadata.get("info", {}).get("requires_dist") or []
-        for recurs_req in recurs_reqs:
+        zip_file = ZipFile(io.BytesIO(wheel_bytes))  # type: ignore
+        dist = pkg_resources_distribution_for_wheel(zip_file, name, "???")
```
Comment (rth) [inline packages/micropip/src/micropip/micropip.py:251]:
```suggestion
        with ZipFile(io.BytesIO(wheel_bytes)) as zip_file:  # type: ignore
            dist = pkg_resources_distribution_for_wheel(zip_file, name, "???")
```

**Review by rth — COMMENTED** [review]
Generally LGTM. +1 to merge this simpler implementation. 

> Yet again install-custom-url is failing.

Hmm, looks like something is different between PyPi and hosting ourselves. Don't have much ideas about it short of spending time debugging..

**Comment by hoodmane** [comment]
Okay it's definitely getting a 404.

**Comment by hoodmane** [comment]
Okay the problem is really simple, I changed the url in the test and for some reason I didn't look at the diff with the test on `main` until just now.

**Comment by hoodmane** [comment]
Should be fixed.

**Pull request merged**